docs/source/troubleshooting/basic_chatops_troubleshooting.rst

```diff
@@ -0,0 +1,107 @@
+ChatOps Troubleshooting Guide
+=============================
+
+This guide will help you troubleshoot ``st2chatops``, up to the point where Hubot should work, reply to your
+commands and send manually triggered messages via ``chatops.post_message``. 
+
+**Note:** Recommended way to install ``st2chatops`` is using the steps mentioned
+in :doc:`Installation Section <../install/index>` for the supported OS.
+
+----------------------------------------------
+Troubleshooting using Hubot self-check script:
+----------------------------------------------
+
+We have a `self-check script <https://github.com/StackStorm/st2chatops/blob/master/scripts/self-check.sh>`_ 
+to help you debug chatops issue in StackStorm 1.5 and above.
+
+Just copy and run it on the server. It runs a few essential tests giving you basic troubleshooting steps in
+case of a failure.
+
+
+--------------------------------------------------
+Manual troubleshooting steps (same as the script):
+--------------------------------------------------
+
+The assumption is that StackStorm client can connect to the server instance. Authentication is in place and
+``st2 run core.local cmd=echo`` succeeds. (This is checked in the script mentioned above.). Following are the
+issues that users usually face with ChatOps:
+
+
+Your bot is not online:
+-----------------------
+
+You've installed StackStorm, but Hubot didn't come online in your ChatOps client (Slack, Hipchat etc..)
+
+    *Possible reasons:*
+
+    - Incorrect adapter settings. Check ``/opt/stackstorm/chatops/st2chatops.env`` for adapter settings.
+      For example, for slack adapter, we need to uncomment following lines and update slack token:
+
+        .. code-block:: shell
+
+           # Slack settings (https://github.com/slackhq/hubot-slack):
+           # export HUBOT_ADAPTER=slack
+           # export HUBOT_SLACK_TOKEN=xoxb-CHANGE-ME-PLEASE
+
+      After changing the adapter settings don't forget to restart the service using this command:
+        
+        .. code-block:: shell
+
+           $ sudo service st2chatops restart
+
+    - Also, make sure the login credentials and ``ST2_HOSTNAME`` are correct.
+    - Check if Hubot-stacksotrm is installed:
+
+        .. code-block:: shell
+
+           $ cd /opt/stackstorm/chatops && npm list | grep hubot-stackstorm
+
+      In case the installation is outdated or corrupt, install hubot-stackstorm:
+      
+        .. code-block:: shell
+        
+           $ cd /opt/stackstorm/chatops && npm install hubot-stackstorm
+
+
+There's no StackStorm commands in `!help`:
+--------------------------------------------
+
+Hubot is online, but when you say ``!help``, only default commands are listed (no commands starting with ``st2``).
```

Comment:
Please include a link to invite a chatops bot into a channel from Internet. People miss this step.